[Bug] Cannot export nose cone with "export to separate files" option

### Describe the bug

Please include a description about what actions you were performing when the exception occurred:

1. Trying to export a nose cone to obj, with the export to separate files option enabled
2. When the separate files option is not enabled, the export works fine

```
---------- Exception stack trace ----------
java.lang.RuntimeException: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1698) at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1685) at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1678) at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) at java.desktop/java.awt.Component.processEvent(Component.java:6391) at java.desktop/java.awt.Container.processEvent(Container.java:2266) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) Caused by: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) at java.base/java.io.FileOutputStream.open0(Native Method) at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) ... 43 more

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 24.12.beta.01
OpenRocket source: default
OpenRocket location: C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\.\jar\OpenRocket.jar
User-defined thrust curves location: C:\Users\ayaan.barbhuiya\AppData\Roaming\OpenRocket\ThrustCurves
LAF: com.formdev.flatlaf.FlatLightLaf
JOGL version: null
Current default locale: en_US
System properties:
apple.awt.application.appearance=system
exe4j.consoleCodepage=cp0
exe4j.isInstall4j=true
exe4j.launchName=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\OpenRocket.exe
exe4j.moduleName=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\OpenRocket.exe
exe4j.semaphoreName=Local\c:_users_ayaan.barbhuiya_appdata_local_openrocket_openrocket.exe0
exe4j.tempDir=
exe4j.unextractedPosition=0
file.encoding=Cp1252
file.separator=\
flatlaf.uiScale=1
flatlaf.uiScale.enabled=true
install4j.appDir=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\
install4j.exeDir=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\
install4j.launcherId=59
install4j.swt=false
java.class.path=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\.install4j\i4jruntime.jar;C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\.\jar\OpenRocket.jar
java.class.version=61.0
java.home=c:\users\ayaan.barbhuiya\appdata\local\openrocket\jre
java.io.tmpdir=C:\Users\AYAAN~1.BAR\AppData\Local\Temp\
java.library.path=c:\users\ayaan.barbhuiya\appdata\local\openrocket\jre\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Windows\CCM;C:\Windows\CCM;C:\Windows\CCM;C:\Users\ayaan.barbhuiya\AppData\Local\Microsoft\WindowsApps;
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.13+12-LTS
java.specification.maintenance.version=1
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.util.Arrays.useLegacyMergeSort=true
java.vendor=BellSoft
java.vendor.url=https://bell-sw.com/
java.vendor.url.bug=https://bell-sw.com/support
java.version=17.0.13
java.version.date=2024-10-15
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=BellSoft
java.vm.version=17.0.13+12-LTS
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln10577678132775008137
line.separator=\u000d\u000a
native.encoding=Cp1252
os.arch=amd64
os.name=Windows 10
os.version=10.0
path.separator=;
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=info.openrocket.swing.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=c:\users\ayaan.barbhuiya\appdata\local\openrocket\jre\bin
sun.cpu.endian=little
sun.cpu.isalist=amd64
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket\OpenRocket.exe
sun.jnu.encoding=Cp1252
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.os.patch.level=
user.country=US
user.dir=C:\Users\ayaan.barbhuiya\AppData\Local\OpenRocket
user.home=C:\Users\ayaan.barbhuiya
user.language=en
user.name=Ayaan.Barbhuiya
user.script=
user.timezone=America/Los_Angeles
user.variant=

---------- Error log ----------
[LOG CUT DOWN DUE TO SIZE RESTIRCTIONS]
325242 1352.360 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Nose Cone' to info.openrocket.core.document.OpenRocketDocument@3bac07a6, document is in unclean state
325416 0.000 WARN (-) ===== 977 WARN lines removed =====
325416 1354.228 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@f62afb6 removing listener that was not registered: javax.swing.JSlider$ModelListener@4a56f60a
325418 1354.228 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@2ce2a435 but it was not registered
325423 1354.228 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@191ba2bf while still having listeners [javax.swing.JSlider$ModelListener@7bb2a2de, javax.swing.plaf.basic.BasicSliderUI$Handler@195c13bc]
325427 1354.228 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@191ba2bf removing listener that was not registered: javax.swing.JSlider$ModelListener@7bb2a2de
325429 1354.228 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7cb56152 but it was not registered
325434 1354.228 WARN (ModelInvalidator.java:111) Attempted to remove ComponentChangeListener info.openrocket.swing.gui.adaptors.PresetModel@6079e1cf but it was not registered
325446 1354.228 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4992eef0 but it was not registered
325458 1354.228 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@12258002 but it was not registered
325470 1354.228 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@22fc446f but it was not registered
325511 0.000 VBOSE (-) ===== 319785 VBOSE lines removed =====
325511 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 3
325514 1354.229 WARN (BooleanModel.java:352) Invalidating BooleanModel[internal value] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@7c352e34]
325515 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@7f86afe4
325518 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@15fdb3f8
325519 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@15fdb3f8 while still having listeners [javax.swing.JSpinner$ModelListener@6d49a05]
325520 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325523 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1842ddbb
325524 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1842ddbb while still having listeners [javax.swing.JSpinner$ModelListener@3ae46804]
325525 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325528 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@64affd92
325529 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@64affd92 while still having listeners [javax.swing.JSpinner$ModelListener@7d920d4b, javax.swing.JSlider$ModelListener@39b167ab, javax.swing.plaf.basic.BasicSliderUI$Handler@21740fd2]
325530 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325533 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@240b8104
325534 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@240b8104 while still having listeners [javax.swing.JSpinner$ModelListener@7861ddd0]
325535 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325538 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@26cedbff
325539 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@26cedbff while still having listeners [javax.swing.JSpinner$ModelListener@50b70dab]
325540 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325543 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1d2471d7
325544 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1d2471d7 while still having listeners [javax.swing.JSpinner$ModelListener@7596fd8d, javax.swing.JSlider$ModelListener@4265a196, javax.swing.plaf.basic.BasicSliderUI$Handler@2532fb0f]
325545 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325547 0.000 DEBUG (-) ===== 3601 DEBUG lines removed =====
325547 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@1d2471d7
325548 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2b2c9b2b
325549 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2b2c9b2b while still having listeners [javax.swing.JSpinner$ModelListener@11eae3d, javax.swing.JSlider$ModelListener@3f773dca, javax.swing.plaf.basic.BasicSliderUI$Handler@4b6c135f]
325550 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325551 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[AppearanceBuilder:Rotation]
325552 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@2b2c9b2b
325553 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@76e1e7db
325554 1354.229 VBOSE (ModelInvalidator.java:109) Removing change listeners, listener count is now 2
325555 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.appearance.AppearanceBuilder:EdgeMode]
325556 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@76e1e7db
325557 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@df049a0
325558 1354.229 WARN (ModelInvalidator.java:111) Attempted to remove ComponentChangeListener info.openrocket.swing.gui.adaptors.TransitionShapeModel@3c4b92f2 but it was not registered
325559 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: info.openrocket.swing.gui.adaptors.TransitionShapeModel@3c4b92f2
325560 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@df049a0
325561 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1ef03e04
325562 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7eccff2d but it was not registered
325563 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShapeParameter]
325564 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@1ef03e04
325565 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@b6eb8b3
325566 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@b6eb8b3 while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1058de76]
325567 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@5056e167, current number of listeners is 76
325568 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShapeParameterMin]
325569 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@b6eb8b3
325570 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@77e258c3
325571 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@77e258c3 while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1058de76]
325572 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4fc9f577, current number of listeners is 75
325573 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShapeParameterMax]
325574 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@77e258c3
325575 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@6845bacb
325576 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4e55592b but it was not registered
325577 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:Length]
325578 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@6845bacb
325579 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@5a22a6b4
325580 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@6fc0533c but it was not registered
325581 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:BaseRadius]
325582 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@5a22a6b4
325583 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@3d600b5c
325584 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@2760b143 but it was not registered
325585 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:Thickness]
325586 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@3d600b5c
325587 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@5cc89719
325588 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7bd31f89, current number of listeners is 74
325589 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[NoseCone:Filled]
325590 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@5cc89719
325591 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@959463a
325592 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@2e543a67, current number of listeners is 73
325593 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[NoseCone:Flipped]
325594 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@959463a
325595 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@6a33fc3d
325596 1354.229 WARN (ModelInvalidator.java:111) Attempted to remove ComponentChangeListener info.openrocket.swing.gui.adaptors.MaterialModel@5120c599 but it was not registered
325597 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: info.openrocket.swing.gui.adaptors.MaterialModel@5120c599
325598 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@6a33fc3d
325599 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@6a092c58
325600 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@137e0156 but it was not registered
325601 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.NoseCone:Finish]
325602 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@6a092c58
325603 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@f064a79
325604 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@f064a79 while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@46196611, info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@6ed34300]
325605 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[constant=0.0]
325606 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@f064a79
325607 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@3d7184c4
325608 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@6eb7b636 but it was not registered
325609 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShoulderRadius]
325610 1354.229 DEBUG (ModelInvalidator.java:113) 1997 of 1997 objects remaining in discarded objects list after purge.
325611 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@3d7184c4
325612 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@740ffd7f
325613 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@740ffd7f while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@46196611]
325614 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@72973488, current number of listeners is 72
325615 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:BaseRadius]
325616 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@740ffd7f
325617 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@f62afb6
325618 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@2ce2a435 but it was not registered
325619 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShoulderLength]
325620 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@f62afb6
325621 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@191ba2bf
325622 1354.229 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7cb56152 but it was not registered
325623 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShoulderThickness]
325624 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@191ba2bf
325625 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@5c6fd8fc
325626 1354.229 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@5c6fd8fc while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@6ed34300]
325627 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@5128b0c6, current number of listeners is 71
325628 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[NoseCone:ShoulderRadius]
325629 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@5c6fd8fc
325630 1354.229 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@43ea317a
325631 1354.229 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4f6915bc, current number of listeners is 70
325632 1354.229 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[NoseCone:ShoulderCapped]
325633 1354.229 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@43ea317a
325634 1354.229 VBOSE (ComponentConfigDialog.java:76) Removed ComponentChangeListener info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog14,54,36,773x453,invalid,hidden,layout=java.awt.BorderLayout,MODELESS,title=Nose Cone configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,0,761x447,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true], current number of listeners is 69
325635 1358.070 DEBUG (FileHelper.java:149) File name does not contain extension, adding 'obj'
325636 1358.070 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6b81bb5f adding listener (total 1): javax.swing.JSpinner$ModelListener@2e538ec3
325637 1358.070 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6b81bb5f adding listener (total 2): info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@63d08494
325638 1358.070 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6b81bb5f adding listener (total 3): info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@4a1490d1
325639 1358.070 DEBUG (DoubleModel.java:873) Setting value 1000.0 for DoubleModel[OBJExportOptions:ScalingDouble]
325640 1376.659 ERROR (SwingExceptionHandler.java:49) Handling uncaught exception on thread=Thread[AWT-EventQueue-0,6,main]
> java.lang.RuntimeException: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) > at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1698) > at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1685) > at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1678) > at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274) > at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) > at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) > at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) > at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) > at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) > at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) > at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) > at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) > at java.desktop/java.awt.Component.processEvent(Component.java:6391) > at java.desktop/java.awt.Container.processEvent(Container.java:2266) > at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) > at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) > at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) > at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) > at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) > at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) > Caused by: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) > at java.base/java.io.FileOutputStream.open0(Native Method) > at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) > ... 43 more
325693 1376.659 INFO (SwingExceptionHandler.java:62) Exception handler running on EDT, showing dialog
325694 1376.659 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
325695 1378.895 USER (SwingExceptionHandler.java:221) User requested sending bug report
325696 1378.898 DEBUG (BugReportDialog.java:228) Found jar file using codeSource
325697 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[OBJExportOptions:ScalingDouble] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@2e538ec3, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@63d08494, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@4a1490d1]
325698 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[OBJExportOptions:ScalingDouble] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@2e538ec3, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@63d08494, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@4a1490d1]
325699 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.7] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@5f5e8ce5]
325700 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.7] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@5f5e8ce5]
325701 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.2] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@6382043e]
325702 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.2] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@6382043e]
325703 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.2] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@5a5474b4]
325704 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.2] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@5a5474b4]
325705 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=1.0] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@32bc5c19]
325706 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.0635] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1e53171c]
325707 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.0635] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1e53171c]
325708 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.0] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1e53171c]
325709 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=0.0] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1e53171c]
325710 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[SVGOptionPanel:StrokeWidth] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@431f0c4c]
325711 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[SVGOptionPanel:StrokeWidth] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@431f0c4c]
325712 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[OBJExportOptions:ScalingDouble] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@d86e91b, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@64b5f9cf, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@168a580d]
325713 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[OBJExportOptions:ScalingDouble] being garbage-collected while having listeners [javax.swing.JSpinner$ModelListener@d86e91b, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@64b5f9cf, info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@168a580d]
325714 1379.009 WARN (ModelInvalidator.java:121) DoubleModel[constant=1.0] being garbage-collected while having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@32bc5c19]
325715 1447.286 DEBUG (GUIUtil.java:279) Clearing all models of window info.openrocket.swing.gui.dialogs.BugReportDialog[dialog15,98,98,793x625,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Bug report,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,0,781x619,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
325716 1447.286 DEBUG (GUIUtil.java:281) Adding object into collectable list: info.openrocket.swing.gui.dialogs.BugReportDialog[dialog15,98,98,793x625,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Bug report,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,6,0,781x619,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
325717 1454.821 DEBUG (FileHelper.java:149) File name does not contain extension, adding 'obj'
325718 1454.823 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6897f911 adding listener (total 1): javax.swing.JSpinner$ModelListener@f0fa5d7
325719 1454.823 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6897f911 adding listener (total 2): info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$10@5d8bdc87
325720 1454.823 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@6897f911 adding listener (total 3): info.openrocket.swing.file.wavefrontobj.OBJOptionChooser$7@7850bcac
325721 1454.825 DEBUG (DoubleModel.java:873) Setting value 1000.0 for DoubleModel[OBJExportOptions:ScalingDouble]
325722 1464.384 ERROR (SwingExceptionHandler.java:49) Handling uncaught exception on thread=Thread[AWT-EventQueue-0,6,main]
> java.lang.RuntimeException: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) > at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1698) > at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1685) > at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1678) > at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274) > at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) > at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) > at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) > at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) > at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) > at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) > at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) > at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) > at java.desktop/java.awt.Component.processEvent(Component.java:6391) > at java.desktop/java.awt.Container.processEvent(Container.java:2266) > at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) > at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) > at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) > at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) > at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) > at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) > Caused by: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied) > at java.base/java.io.FileOutputStream.open0(Native Method) > at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) > ... 43 more
325725 0.000 STDERR (-) ===== 806 STDERR lines removed =====
325725 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220)
325726 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1698)
325727 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1685)
325728 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1678)
325729 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274)
325730 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
325731 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
325732 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
325733 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
325734 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374)
325735 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028)
325736 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072)
325737 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
325738 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
325739 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
325740 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.processEvent(Component.java:6391)
325741 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.processEvent(Container.java:2266)
325742 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
325743 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
325744 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
325745 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
325746 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
325747 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
325748 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
325749 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
325750 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
325751 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775)
325752 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
325753 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
325754 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
325755 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
325756 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
325757 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747)
325758 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
325759 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
325760 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
325761 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744)
325762 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
325763 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
325764 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
325765 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
325766 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
325767 1464.384 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
325768 1464.384 STDERR (SwingExceptionHandler.java:50) Caused by: java.io.FileNotFoundException: C:\Users\ayaan_1_Nose Cone D.obj (Access is denied)
325769 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.open0(Native Method)
325770 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293)
325771 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235)
325772 1464.384 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155)
325773 1464.384 STDERR (SwingExceptionHandler.java:50) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230)
325774 1464.384 STDERR (SwingExceptionHandler.java:50) ... 43 more
325775 1464.384 INFO (SwingExceptionHandler.java:62) Exception handler running on EDT, showing dialog
325776 1464.384 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
325777 1466.792 USER (SwingExceptionHandler.java:221) User requested sending bug report
325778 1466.792 DEBUG (BugReportDialog.java:228) Found jar file using codeSource
```

### To Reproduce

1. Open the ork file
2. Export the nose cone as an obj, with the export to seperate files option enabled


### Screenshots / .ork file

#### Screenshot(s):

![Image](https://github.com/user-attachments/assets/0fe23d66-1733-452e-bfe9-587c53f24b07)

#### .ork file:

[Rocket Design Template 2025.zip](https://github.com/user-attachments/files/18649139/Rocket.Design.Template.2025.zip)

### OpenRocket version

24.12.beta.01

### What platform are you running on?

Windows

### Additional context

_No response_

--- a/core/src/main/java/info/openrocket/core/util/FileUtils.java
+++ b/core/src/main/java/info/openrocket/core/util/FileUtils.java
@@ -44,36 +44,10 @@ public abstract class FileUtils {
 	 * @return the file name without extension
 	 */
 	public static String removeExtension(String fileName) {
-		if (fileName == null || fileName.isEmpty()) {
-			return fileName;
-		}
-
-		// First get just the filename portion if this is a path
-		String name = Paths.get(fileName).getFileName().toString();
-
-		// Handle special case of files starting with a dot
-		if (name.startsWith(".")) {
-			if (name.length() == 1) { // Just a dot
-				return name;
-			}
-			// Check if there's another dot in the name after the first character
-			int secondDot = name.indexOf('.', 1);
-			if (secondDot == -1) { // No other dots, it's a dotfile like .gitignore
-				return "";
-			}
-			// Has another extension after the dot prefix
-			return name.substring(0, secondDot);
-		}
-
-		// Find last occurrence of dot in the filename
-		int lastDot = name.lastIndexOf('.');
-
-		if (lastDot == -1) { // No extension found
-			return name;
-		}
-
-		return name.substring(0, lastDot);
+		return fileName == null || fileName.isEmpty() ? fileName :
+				fileName.replaceFirst("[.][^.]+$", "");
 	}
+
 
 	/**
 	 * Get the file name from a path.

--- a/core/src/main/java/info/openrocket/core/util/FileUtils.java
+++ b/core/src/main/java/info/openrocket/core/util/FileUtils.java
@@ -38,17 +38,41 @@ public abstract class FileUtils {
 	}
 
 	/**
-	 * Remove the extension from a file name.
-	 * 
-	 * @param fileName the file name
+	 * Remove the extension from a file name, properly handling paths and edge cases.
+	 *
+	 * @param fileName the file name or path
 	 * @return the file name without extension
 	 */
 	public static String removeExtension(String fileName) {
-		String[] splitResults = fileName.split("\\.");
-		if (splitResults.length > 0) {
-			return splitResults[0];
+		if (fileName == null || fileName.isEmpty()) {
+			return fileName;
 		}
-		return fileName;
+
+		// First get just the filename portion if this is a path
+		String name = Paths.get(fileName).getFileName().toString();
+
+		// Handle special case of files starting with a dot
+		if (name.startsWith(".")) {
+			if (name.length() == 1) { // Just a dot
+				return name;
+			}
+			// Check if there's another dot in the name after the first character
+			int secondDot = name.indexOf('.', 1);
+			if (secondDot == -1) { // No other dots, it's a dotfile like .gitignore
+				return "";
+			}
+			// Has another extension after the dot prefix
+			return name.substring(0, secondDot);
+		}
+
+		// Find last occurrence of dot in the filename
+		int lastDot = name.lastIndexOf('.');
+
+		if (lastDot == -1) { // No extension found
+			return name;
+		}
+
+		return name.substring(0, lastDot);
 	}
 
 	/**

--- a/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
+++ b/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
@@ -107,6 +107,7 @@ public class FileUtilsTest {
 		// Test periods in path
 		assertEquals("test", FileUtils.removeExtension("/path/with.dots/test.txt"));
 		assertEquals("test", FileUtils.removeExtension("C:/my.folder/sub.folder/test.txt"));
+		assertEquals("test.1.2", FileUtils.removeExtension("C:/my.folder/sub.folder/test.1.2.txt"));
 		assertEquals("test", FileUtils.removeExtension("folder.name/test.txt"));
 	}
 

--- a/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
+++ b/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
@@ -1,0 +1,183 @@
+package info.openrocket.core.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class FileUtilsTest {
+	@TempDir
+	Path tempDir;
+
+	private File testFile;
+	private static final String TEST_CONTENT = "Hello, World!";
+
+	@BeforeEach
+	void setUp() throws IOException {
+		testFile = tempDir.resolve("test.txt").toFile();
+		try (FileWriter writer = new FileWriter(testFile)) {
+			writer.write(TEST_CONTENT);
+		}
+	}
+
+	@Test
+	void testCopy_WithBufferedStreams() throws IOException {
+		try (InputStream is = new BufferedInputStream(new FileInputStream(testFile));
+			 ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+
+			FileUtils.copy(is, os);
+
+			assertEquals(TEST_CONTENT, os.toString());
+		}
+	}
+
+	@Test
+	void testCopy_WithUnbufferedStreams() throws IOException {
+		try (InputStream is = new FileInputStream(testFile);
+			 ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+
+			FileUtils.copy(is, os);
+
+			assertEquals(TEST_CONTENT, os.toString());
+		}
+	}
+
+	@Test
+	void testCopy_WithEmptyFile() throws IOException {
+		File emptyFile = tempDir.resolve("empty.txt").toFile();
+		// Create the empty file
+		assertTrue(emptyFile.createNewFile());
+
+		try (InputStream is = new FileInputStream(emptyFile);
+			 ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+
+			FileUtils.copy(is, os);
+
+			assertEquals(0, os.size());
+		}
+	}
+
+	@Test
+	void testReadBytes() throws IOException {
+		try (InputStream is = new FileInputStream(testFile)) {
+			byte[] bytes = FileUtils.readBytes(is);
+
+			assertEquals(TEST_CONTENT, new String(bytes));
+		}
+	}
+
+	@Test
+	void testReadBytes_EmptyFile() throws IOException {
+		File emptyFile = tempDir.resolve("empty.txt").toFile();
+		// Create the empty file
+		assertTrue(emptyFile.createNewFile());
+
+		try (InputStream is = new FileInputStream(emptyFile)) {
+			byte[] bytes = FileUtils.readBytes(is);
+
+			assertEquals(0, bytes.length);
+		}
+	}
+
+	@Test
+	void testRemoveExtension_NormalFile() {
+		assertEquals("filename", FileUtils.removeExtension("filename.txt"));
+	}
+
+	@Test
+	void testRemoveExtension_NoExtension() {
+		assertEquals("filename", FileUtils.removeExtension("filename"));
+	}
+
+	@Test
+	void testRemoveExtension_DotFile() {
+		assertEquals("", FileUtils.removeExtension(".gitignore"));
+		// Test periods in path
+		assertEquals("test", FileUtils.removeExtension("/path/with.dots/test.txt"));
+		assertEquals("test", FileUtils.removeExtension("C:/my.folder/sub.folder/test.txt"));
+		assertEquals("test", FileUtils.removeExtension("folder.name/test.txt"));
+	}
+
+	@Test
+	void testGetFileNameFromPath() {
+		// Test absolute paths
+		assertEquals("file.txt", FileUtils.getFileNameFromPath("/path/to/file.txt"));
+		assertEquals("document.pdf", FileUtils.getFileNameFromPath("C:/Documents/document.pdf"));
+
+		// Test relative paths
+		assertEquals("image.jpg", FileUtils.getFileNameFromPath("../folder/image.jpg"));
+		assertEquals("script.sh", FileUtils.getFileNameFromPath("./script.sh"));
+
+		// Test just filename
+		assertEquals("simple.txt", FileUtils.getFileNameFromPath("simple.txt"));
+
+		// Test paths with spaces and special characters
+		assertEquals("my file.txt", FileUtils.getFileNameFromPath("/path/to/my file.txt"));
+		assertEquals("file with spaces.doc", FileUtils.getFileNameFromPath("folder/file with spaces.doc"));
+	}
+
+	@Test
+	void testGetIllegalFilenameChar_ValidFilename() {
+		assertNull(FileUtils.getIllegalFilenameChar("valid-filename.txt"));
+	}
+
+	@Test
+	void testGetIllegalFilenameChar_IllegalChars() {
+		assertEquals('/', FileUtils.getIllegalFilenameChar("illegal/filename"));
+		assertEquals('\\', FileUtils.getIllegalFilenameChar("illegal\\filename"));
+		assertEquals(':', FileUtils.getIllegalFilenameChar("illegal:filename"));
+		assertEquals('*', FileUtils.getIllegalFilenameChar("illegal*filename"));
+		assertEquals('?', FileUtils.getIllegalFilenameChar("illegal?filename"));
+		assertEquals('"', FileUtils.getIllegalFilenameChar("illegal\"filename"));
+		assertEquals('<', FileUtils.getIllegalFilenameChar("illegal<filename"));
+		assertEquals('>', FileUtils.getIllegalFilenameChar("illegal>filename"));
+		assertEquals('|', FileUtils.getIllegalFilenameChar("illegal|filename"));
+	}
+
+	@Test
+	void testGetIllegalFilenameChar_EmptyString() {
+		assertNull(FileUtils.getIllegalFilenameChar(""));
+	}
+
+	@Test
+	void testGetIllegalFilenameChar_NullInput() {
+		assertNull(FileUtils.getIllegalFilenameChar(null));
+	}
+
+	@Test
+	void testCopy_WithClosedInputStream() {
+		assertThrows(IOException.class, () -> {
+			InputStream is = new FileInputStream(testFile);
+			is.close();
+
+			try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+				FileUtils.copy(is, os);
+			}
+		});
+	}
+
+	@Test
+	void testCopy_WithClosedOutputStream() {
+		File outputFile = tempDir.resolve("output.txt").toFile();
+		assertThrows(IOException.class, () -> {
+			try (InputStream is = new FileInputStream(testFile)) {
+				FileOutputStream os = new FileOutputStream(outputFile);
+				os.close();
+
+				FileUtils.copy(is, os);
+			}
+		});
+	}
+}

--- a/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
+++ b/core/src/test/java/info/openrocket/core/util/FileUtilsTest.java
@@ -105,10 +105,10 @@ public class FileUtilsTest {
 	void testRemoveExtension_DotFile() {
 		assertEquals("", FileUtils.removeExtension(".gitignore"));
 		// Test periods in path
-		assertEquals("test", FileUtils.removeExtension("/path/with.dots/test.txt"));
-		assertEquals("test", FileUtils.removeExtension("C:/my.folder/sub.folder/test.txt"));
-		assertEquals("test.1.2", FileUtils.removeExtension("C:/my.folder/sub.folder/test.1.2.txt"));
-		assertEquals("test", FileUtils.removeExtension("folder.name/test.txt"));
+		assertEquals("/path/with.dots/test", FileUtils.removeExtension("/path/with.dots/test.txt"));
+		assertEquals("C:/my.folder/sub.folder/test", FileUtils.removeExtension("C:/my.folder/sub.folder/test.txt"));
+		assertEquals("C:/my.folder/sub.folder/test.1.2", FileUtils.removeExtension("C:/my.folder/sub.folder/test.1.2.txt"));
+		assertEquals("folder.name/test", FileUtils.removeExtension("folder.name/test.txt"));
 	}
 
 	@Test

